tempesta_fw/procfs.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __TFW_PROCFS_H__
+#define __TFW_PROCFS_H__
+
+#include "tempesta_fw.h"
+
+/*
+ * @msgs_received	- Total number of messages received from a peer.
+ * @msgs_forwarded	- The number of messages forwarded to the other side.
+ * @msgs_notparsed	- The number of messages with parsing errors.
+ * @msgs_filterout	- The number of messages that were filtered out
+ *			  in accordance with the rules in configuration.
+ * @msgs_otherrors	- The number of messages not accepted due to
+ *			  other errors.
+ *
+ * @connects		- Total number of connect attempts.
+ * @disconnects		- Total number of disconnects for any reason.
+ * @connections		- Total number of connections ever established
+ *			  with peers while Tempesta is active.
+ *
+ * @bytes_received	- Total number of bytes received from a peer and
+ *			  processed by Tempesta.
+ *
+ * Note: unsigned long type is 64 bits on 64-bit architecture.
+ */
+struct tfw_peer_stats {
+	unsigned long	msgs_received;
+	unsigned long	msgs_forwarded;
+	unsigned long	msgs_notparsed;
+	unsigned long	msgs_filterout;
+	unsigned long	msgs_otherrors;
+
+	unsigned long	connects;
+	unsigned long	disconnects;
+	unsigned long	connections;
+
+	unsigned long	bytes_received;
+};
+
+typedef struct tfw_perfstat {
+	struct tfw_peer_stats	clnt;
+	struct tfw_peer_stats	serv;
+} tfw_perfstat_t;
+
+DECLARE_PER_CPU_ALIGNED(tfw_perfstat_t, tfw_perfstat);
+
+#define TFW_INC_STAT_BH(...)	__this_cpu_inc(tfw_perfstat.__VA_ARGS__)
+#define TFW_ADD_STAT_BH(val, ...)	\
+		__this_cpu_add(tfw_perfstat.__VA_ARGS__, val)
```

Comment:
Do we really need preemption check? Typically the code is running in softirq in recent master branch.